test/parser.js

```diff
@@ -0,0 +1,77 @@
+import { expect } from 'chai';
+import fury from 'fury';
+import Parser from '../src/parser';
+
+describe('Parser', () => {
+  let parser;
+
+  before(() => {
+    parser = new Parser({ minim: fury.minim });
+    parser.swagger = {
+      consumes: [],
+      produces: [],
+    };
+  });
+
+  context('content types', () => {
+    it('gathers null response type when no examples or produces', () => {
+      const methodValue = {};
+      const examples = {};
+      const contentTypes = parser.gatherResponseContentTypes(methodValue, examples);
+
+      expect(contentTypes).to.deep.equal([null]);
+    });
+
+    it('gathers all example response content types', () => {
+      const methodValue = {};
+      const examples = {
+        'application/json': '',
+        'application/hal+json': '',
+        'application/xml': '',
+      };
+      const contentTypes = parser.gatherResponseContentTypes(methodValue, examples);
+
+      expect(contentTypes).to.deep.equal([
+        'application/json',
+        'application/hal+json',
+        'application/xml',
+      ]);
+    });
+
+    it('gathers first JSON produces without examples', () => {
+      const methodValue = {
+        produces: [
+          'application/json',
```

Comment:
I'd prepend the array with one non-JSON content type so it's clear those would be skipped.

Reply:
Added in c86d5bc